tests/specs/mcd/verification.k

```diff
@@ -0,0 +1,77 @@
+requires "../lemmas.k"
+requires "bin_runtime.k"
+requires "storage.k"
+
+module HASKELL-HASH2 [symbolic, kore]
+    imports EDSL
+
+    syntax Int ::= hash2(Int, Int) [function]
+ // -----------------------------------------
+endmodule
+
+module VERIFICATION
+    imports LEMMAS
+    imports DSS-BIN-RUNTIME
+    imports DSS-STORAGE
+    imports HASKELL-HASH2
+
+    syntax Bool ::= #notPrecompileAddress ( Int ) [function]
+ // --------------------------------------------------------
+    rule #notPrecompileAddress ( X ) => 0 ==Int X orBool (10 <=Int X andBool #rangeAddress(X))
+
+    // arithmetic/data-structures
+    rule I modInt N => I requires 0 <=Int I andBool I <Int N
+    rule I in (SetItem(I') REST) => I in REST requires I =/=Int I'
+
+    // chop
+    rule chop(hash2(V1, V2) +Int 1) => hash2(V1, V2) +Int 1
+
+    // sizeWordStack/sizeByteArray
+    rule #sizeByteArray(WS ++ WS')                        => #sizeByteArray(WS) +Int #sizeByteArray(WS')
+    rule #sizeByteArray(#padToWidth(32, #asByteStack(I))) => 32 requires 0 <=Int I andBool I <Int pow256
+    rule #sizeByteArray(_WS [ _START .. WIDTH ])          => WIDTH
+
+    rule 0 <=Int #sizeWordStack(_, _) => true [smt-lemma]
+    rule 0 <=Int #sizeByteArray(_)    => true [smt-lemma]
+
+    // ranged lookups in ByteArray
+    rule (WS ++ WS') [ 0 .. WIDTH ] => WS ++ (WS' [ 0 .. WIDTH -Int #sizeByteArray(WS) ]) requires #sizeByteArray(WS) <=Int WIDTH
+
+    rule (WS ++ _WS') [ START .. WIDTH ] => WS  [ START                         .. WIDTH ] requires START +Int WIDTH <=Int #sizeByteArray(WS)
+    rule (WS ++  WS') [ START .. WIDTH ] => WS' [ START -Int #sizeByteArray(WS) .. WIDTH ] requires #sizeByteArray(WS) <=Int START
```

Comment:
in more rigurous form, those are present in `LEMMAS-HASKELL` : https://github.com/kframework/evm-semantics/blob/3fc250d96249feefadbd3c5785875dd2bae5a934/tests/specs/lemmas.k#L266-L271
Can be moved to `LEMMAS`.

Reply:
Removed.